tests/vmi_networking_test.go

```diff
@@ -879,3 +885,21 @@ func gatewayIPFromCIDR(cidr string) string {
 	ip[oct]++
 	return ip.String()
 }
+
+func isDualStack() bool {
```

Comment:
Why dont you check if a cluster supports dual-stack by checking if one of kube-system pods have IPv4 and IPv6 addresses?
Eventually all providers will support dual-stack

Reply:
because if there will be a bug in the provider, the above method wont find it.
Also the check i added do support future providers, starting with 1.18
when all the lanes that are supported, support dual stack we would be able to just Expect that there are 2 IPS,
moreover, there are providers which arent k8s, like kind, for those we dont need to enforce 2, and we dont.

We should first focus on the effort to make dual stack works, imo its good enough for now.

Reply:
> because if there will be a bug in the provider, the above method wont find it.
> Also the check i added do support future providers, starting with 1.18
> when all the lanes that are supported, support dual stack we would be able to just Expect that there are 2 IPS,
> (there are providers which arent k8s, like kind, for those we dont need to enforce 2, and we dont)

Kubevirt tests should not take in account that there might be a bug in the provider, all tests assume the environment is healthy
If there is a bug in the provider the current state of this function wont catch it neither.

> We should first focus on the effort to make dual stack works, imo its good enough for now.
No dout we need to make it work, but this PR is about testing VM's with IPv6

Reply:
i dont agree, bugs can happen in the provider, which will stop making it dual stack, and other stuff wont catch it.
i think we should keep it simple, straight to the point and generic, yet powerful

Reply:
Even though you dropped this commit we will tackle it later so it  is important for me to mention that when you write a test or refactoring one you dont need to worry about the environemnt you need to assume the cluster is functioning with no bugs.
If you want to keep it simple then just check if a pod has both ipv4 and ipv6 addresses, this way you dont bound to any provider.